snapcraft/internal/sources/_base.py

```diff
@@ -62,3 +65,35 @@ def download(self):
             request.raise_for_status()
 
             download_requests_stream(request, self.file)
+
+    def verify_checksum(source_checksum, checkfile):
```

Comment:
Also, is this intended to be a method on `FileBase`? It shouldn't be, and, well, it's not (no `self`), but then it's used as if it is. It should probably be in `sources/__init__.py`.